src/main/java/duke/Task.java

```diff
@@ -0,0 +1,187 @@
+            LocalDate possibleDeadlineDate = Parser.parseDate(possibleDateTime);
+            if (possibleDeadlineDate != null) {
+                this.isDate = true;
+                this.deadlineDate = possibleDeadlineDate;
+                this.deadlineDateString = Parser.parseStringDate(possibleDeadlineDate);
+            }
+            LocalTime possibleDeadlineTime = Parser.parseTime(possibleDateTime);
+            if (possibleDeadlineTime != null) {
+                this.isTime = true;
+                this.deadlineTime = possibleDeadlineTime;
+                this.deadlineTimeString = Parser.parseStringTime(possibleDeadlineTime);
+            }
+        }
+
+        @Override
+        public String toString() {
+            if (isDate && isTime) {
+                String name = super.isDone ? "[D][X] " + task + " (by: " + this.deadlineDateString + " "
+                        + this.deadlineTimeString +")" : "[D][ ] " + task + " (by: " + this.deadlineDateString + " "
+                        + this.deadlineTimeString + ")";
+                return name;
+            } else if (isDate) {
+                String name = super.isDone ? "[D][X] " + task + " (by: " + this.deadlineDateString + " "
+                        +")" : "[D][ ] " + task + " (by: " + this.deadlineDateString + ")";
+                return name;
+            } else if (isTime) {
+                String name = super.isDone ? "[D][X] " + task + " (by: " + this.deadlineTimeString + " "
+                        +")" : "[D][ ] " + task + " (by: " + this.deadlineTimeString + ")";
+                return name;
+            }
+            String name = super.isDone ? "[D][X] " + task + " (by: " + this.deadline +")" : "[D][ ] " + task + " (by: " + this.deadline +")";
+            return name;
+        }
+    }
+
+    private class Event extends Task {
+        private static final long serialVersionUID = 6827458563349815922L;
+        private String task;
+        private String from;
+        private String to;
+        private Boolean isFromDate = false;
+        private Boolean isFromTime = false;
+        private Boolean isToDate = false;
+        private Boolean isToTime = false;
+        private LocalDate fromDate;
+        private LocalTime fromTime;
+        private LocalDate toDate;
+        private LocalTime toTime;
+        private String fromDateString;
+        private String fromTimeString;
+        private String toDateString;
+        private String toTimeString;
+
+        public Event(String task) throws DukeExceptions {
+            super();
+            String[] commands = task.split(" /from ");
+            if (commands.length == 1) {
+                throw new DukeExceptions("OOPS!!! Looks like someone forget when the event begins :)\n Please use /from to indicate begin time");
+            }
+            this.task = commands[0];
+            String[] from_to_timeline = commands[1].split(" /to ");
+            if (from_to_timeline.length == 1) {
```

Comment:
I've noticed some of your variable names use snake case, however the coding standards states that variable names must be in camelCase, so do remember to check and update your variable names